FATN_to_USB/fatn_to_usb/main.py

```
@@ -0,0 +1,355 @@
+        if not FATN_parse.FATNurl:
+            print('❌ No Dropbox link found on FATN website')
+            return None
+
+        # Validate URL scheme before following redirect (SSRF prevention)
+        if not is_safe_url(FATN_parse.FATNurl):
+            print(f'❌ Security: Rejected unsafe URL scheme: {FATN_parse.FATNurl}')
+            return None
+
+        # Get actual Dropbox URL from tinyurl
+        print(f'Resolving redirect: {FATN_parse.FATNurl}')
+        FATNdropboxurl = urlopen(FATN_parse.FATNurl, timeout=30)
```

Ruff (0.14.14)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)